assets/scripts/palette/Palette.jsx

```diff
@@ -28,63 +23,54 @@ function Palette (props) {
     }
   }, [])
 
-  function renderPaletteItems () {
-    const segments = getAllSegmentInfoArray()
-
-    // For each segment, set "disabled" property instead that indicates
-    // whether this segment is in a disabled state for this user
-    // Then filter out disabled segments that do not have the
-    // `alwaysShowInPalette` property set to `true`
-    const displayedSegments = segments
-      .map((segment) => {
-        // Accept segments that don't have the `enableWithFlag` property
-        const enabledByDefault = !segment.enableWithFlag
-        // Accept segments with the `enableWithFlag` property, but only if
-        // the flags have that value set to true.
-        const enabledByFlag =
-          segment.enableWithFlag && flags[segment.enableWithFlag].value
+  const segments = getAllSegmentInfoArray()
 
-        return {
-          ...segment,
-          disabled: !(enabledByDefault || enabledByFlag)
-        }
-      })
-      .filter(
-        (segment) =>
-          !segment.disabled || (segment.disabled && segment.alwaysShowInPalette)
-      )
-
-    // Return all enabled segments as an array of SegmentForPalette components.
-    return displayedSegments.map((segment) => {
-      const variant = segment.paletteIcon
-        ? segment.paletteIcon
-        : Object.keys(segment.details).shift()
+  // For each segment, set "disabled" property instead that indicates
+  // whether this segment is in a disabled state for this user
+  // Then filter out disabled segments that do not have the
+  // `alwaysShowInPalette` property set to `true`
+  const displayedSegments = segments
+    .map((segment) => {
+      // Accept segments that don't have the `enableWithFlag` property
+      const enabledByDefault = !segment.enableWithFlag
+      // Accept segments with the `enableWithFlag` property, but only if
```

Comment:
Do we have test coverage for this and the above conditions? I haven't run this locally and am not familiar with the functionality/edge cases it is for, but at first glance it looks like a potential source of bugs/test coverage could help refactor it to be a bit more straightforward. 

I'm realizing that this might be out of scope for this PR since the pr is more concerned with the tooltips

Reply:
Yep - I did refactor the code here, but didn't write new logic (or change how it works). Let's make a note to visit the test coverage on this later.